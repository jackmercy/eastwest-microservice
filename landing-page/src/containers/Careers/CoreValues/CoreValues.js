import { memo, useMemo } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { useMediaQuery } from 'react-responsive';

import BREAKPOINTS from '../../../constants/BREAKPOINTS';

import WebCoreValues from './WebCoreValues';
import MobileCoreValues from './MobileCoreValues';
import useStyles from './coreValues.styles';

const CoreValues = ({ id }) => {
  const intl = useIntl();
  const classes = useStyles();
  const isMobile = useMediaQuery({
    query: `(max-width: ${BREAKPOINTS.maxMd}px)`,
  });
  const coreValues = useMemo(
    () => [
      {
        key: 'strive-for-excellence',
        title: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.01.TITLE',
        }),
        titleTranslationId: 'CAREERS.CORE_VALUE.01.TITLE',
        description: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.01.TEXT',
        }),
        image: '/static/images/careers/core-value-01.png',
        srcSet: [
          '/static/images/careers/core-value-01-mobile.png 500w',
          '/static/images/careers/core-value-01.png 2000w',
        ].join(', '),
      },
      {
        key: 'think-in-solutions',
        title: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.02.TITLE',
        }),
        titleTranslationId: 'CAREERS.CORE_VALUE.02.TITLE',
        description: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.02.TEXT',
        }),
        image: '/static/images/careers/core-value-02.png',
        srcSet: [
          '/static/images/careers/core-value-02-mobile.png 500w',
          '/static/images/careers/core-value-02.png 2000w',
        ].join(', '),
      },
      {
        key: 'be-inclusive',
        title: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.03.TITLE',
        }),
        titleTranslationId: 'CAREERS.CORE_VALUE.03.TITLE',
        description: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.03.TEXT',
        }),
        image: '/static/images/careers/core-value-03.png',
        srcSet: [
          '/static/images/careers/core-value-03-mobile.png 500w',
          '/static/images/careers/core-value-03.png 2000w',
        ].join(', '),
      },
      {
        key: 'work-towards-productivity',
        title: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.04.TITLE',
        }),
        titleTranslationId: 'CAREERS.CORE_VALUE.04.TITLE',
        description: intl.formatMessage({
          id: 'CAREERS.CORE_VALUE.04.TEXT',
        }),
        image: '/static/images/careers/core-value-04.png',
        srcSet: [
          '/static/images/careers/core-value-04-mobile.png 500w',
          '/static/images/careers/core-value-04.png 2000w',
        ].join(', '),
      },
    ],
    [intl],
  );

  return (
    <div id={id} className={classes.root}>
      {!isMobile ? (
        <WebCoreValues data={coreValues} />
      ) : (
        <MobileCoreValues data={coreValues} />
      )}
    </div>
  );
};
CoreValues.propTypes = { id: PropTypes.string.isRequired };

export default memo(CoreValues);
