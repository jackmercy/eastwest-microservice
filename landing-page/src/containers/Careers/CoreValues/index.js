import dynamic from 'next/dynamic';

export default dynamic(() => import('./CoreValues'), { ssr: false });
