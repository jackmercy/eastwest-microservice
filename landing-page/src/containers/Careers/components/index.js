export { default as SectionTitle } from './SectionTitle';
export { default as MenuScrollableLayout } from './MenuScrollableLayout';
