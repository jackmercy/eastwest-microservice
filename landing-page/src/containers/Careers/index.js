import dynamic from 'next/dynamic';

export default dynamic(() => import('./Careers'), { ssr: false });
