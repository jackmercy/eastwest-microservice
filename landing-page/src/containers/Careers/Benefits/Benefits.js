import Link from 'next/link';
import { memo, useMemo } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import classNames from 'classnames';

import { SectionTitle } from '../components';

import useStyles from './benefits.styles';

const Benefits = ({ id }) => {
  const intl = useIntl();
  const classes = useStyles();
  const benefits = useMemo(
    () => [
      {
        key: 'benefit-01',
        title: intl.formatMessage({ id: 'CAREERS.BENEFITS.01.TITLE' }),
        description: intl.formatMessage({
          id: 'CAREERS.BENEFITS.01.DESCRIPTION',
        }),
        image: '/static/images/careers/benefit-01.png',
      },
      {
        key: 'benefit-02',
        title: intl.formatMessage({ id: 'CAREERS.BENEFITS.02.TITLE' }),
        description: intl.formatMessage({
          id: 'CAREERS.BENEFITS.02.DESCRIPTION',
        }),
        image: '/static/images/careers/benefit-02.png',
      },
      {
        key: 'benefit-03',
        title: intl.formatMessage(
          { id: 'CAREERS.BENEFITS.03.TITLE' },
          { icon: <i className="icon-arrow-right" /> },
        ),
        description: intl.formatMessage({
          id: 'CAREERS.BENEFITS.03.DESCRIPTION',
        }),
        image: '/static/images/careers/benefit-03.png',
      },
      {
        key: 'benefit-04',
        title: intl.formatMessage({ id: 'CAREERS.BENEFITS.04.TITLE' }),
        description: intl.formatMessage({
          id: 'CAREERS.BENEFITS.04.DESCRIPTION',
        }),
        image: '/static/images/careers/benefit-04.png',
      },
      {
        key: 'benefit-05',
        title: intl.formatMessage({ id: 'CAREERS.BENEFITS.05.TITLE' }),
        description: intl.formatMessage({
          id: 'CAREERS.BENEFITS.05.DESCRIPTION',
        }),
        image: '/static/images/careers/benefit-05.png',
      },
      {
        key: 'benefit-06',
        title: intl.formatMessage({ id: 'CAREERS.BENEFITS.06.TITLE' }),
        description: intl.formatMessage({
          id: 'CAREERS.BENEFITS.06.DESCRIPTION',
        }),
        image: '/static/images/careers/benefit-06.png',
      },
    ],
    [intl],
  );

  return (
    <>
      <SectionTitle
        id={id}
        translateId="CAREERS.BENEFITS.TITLE"
        className={classes.sectionTitle}
      />
      <p className={classes.text}>
        {intl.formatMessage({ id: 'CAREERS.BENEFITS.TEXT.01' })}
      </p>
      <div className={classes.benefits}>
        {benefits.map((item) => (
          <div key={item.key} className={classes.benefit}>
            <img
              alt={item.title}
              className={classes.benefitImage}
              src={item.image}
            />
            <h4 className={classes.benefitTitle}>{item.title}</h4>
            <p className={classes.benefitDescription}>{item.description}</p>
          </div>
        ))}
      </div>
      <div className={classes.ourOffice}>
        <h4 className={classes.ourOfficeTitle}>
          {intl.formatMessage({ id: 'CAREERS.BENEFITS.OUR_OFFICE.TITLE' })}
        </h4>
        <img
          alt="our office"
          className={classes.ourOfficeImage}
          src="/static/images/careers/our-office.png"
          srcSet={[
            '/static/images/careers/our-office-mobile.png 500w',
            '/static/images/careers/our-office.png 2000w',
          ].join(', ')}
        />
        <div className={classes.ourOfficeIntroWrapper}>
          <div className={classes.ourOfficeIntro}>
            <h4 className={classes.ourOfficeIntroTitle}>
              {intl.formatMessage({
                id: 'CAREERS.BENEFITS.OUR_OFFICE.INTRO.01.TITLE',
              })}
            </h4>
            <p className={classes.ourOfficeDescription}>
              {intl.formatMessage({
                id: 'CAREERS.BENEFITS.OUR_OFFICE.INTRO.01.DESCRIPTION',
              })}
            </p>
            <Link href="/talent-labs" passHref>
              <a
                href="/#"
                className={classNames(
                  classes.ourOfficeDescription,
                  classes.ourOfficeLink,
                )}
                data-test="our-office-link-talent-labs"
              >
                {intl.formatMessage({
                  id: 'CAREERS.BENEFITS.OUR_OFFICE.INTRO.01.LINK',
                })}
                <i className="icon-arrow-right" />
              </a>
            </Link>
          </div>
          <div className={classes.ourOfficeIntro}>
            <h4 className={classes.ourOfficeIntroTitle}>
              {intl.formatMessage({
                id: 'CAREERS.BENEFITS.OUR_OFFICE.INTRO.02.TITLE',
              })}
            </h4>
            <p className={classes.ourOfficeDescription}>
              {intl.formatMessage({
                id: 'CAREERS.BENEFITS.OUR_OFFICE.INTRO.02.DESCRIPTION',
              })}
            </p>
            <Link href="/internship-program" passHref>
              <a
                href="/#"
                className={classNames(
                  classes.ourOfficeDescription,
                  classes.ourOfficeLink,
                )}
                data-test="our-office-link-internship-program"
              >
                {intl.formatMessage({
                  id: 'CAREERS.BENEFITS.OUR_OFFICE.INTRO.02.LINK',
                })}
                <i className="icon-arrow-right" />
              </a>
            </Link>
          </div>
        </div>
      </div>
    </>
  );
};
Benefits.propTypes = { id: PropTypes.string.isRequired };

export default memo(Benefits);
