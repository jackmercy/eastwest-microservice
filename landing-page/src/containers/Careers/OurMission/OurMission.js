import { memo } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';

import { SectionTitle } from '../components';

import useStyles from './ourMission.styles';

const OurMission = ({ id }) => {
  const intl = useIntl();
  const classes = useStyles();

  return (
    <>
      <SectionTitle id={id} translateId="CAREERS.OUR_MISSION.TITLE" />
      <p className={classes.text}>
        {intl.formatMessage({ id: 'CAREERS.OUR_MISSION.TEXT.01' })}
      </p>
      <img
        alt="our mission"
        className={classes.image}
        src="/static/images/careers/our-mission.png"
      />
      <p className={classes.text}>
        {intl.formatMessage({ id: 'CAREERS.OUR_MISSION.TEXT.02' })}
      </p>
    </>
  );
};
OurMission.propTypes = { id: PropTypes.string.isRequired };

export default memo(OurMission);
