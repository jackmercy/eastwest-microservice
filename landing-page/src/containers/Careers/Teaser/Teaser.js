import { memo, useCallback } from 'react';
import { useIntl } from 'react-intl';
import PropTypes from 'prop-types';

import { Button } from '../../../components';
import { Content } from '../../components';
import { scrollToElement } from '../../../utils';

import useStyles from './teaser.styles';

const Teaser = ({ CTAId }) => {
  const intl = useIntl();
  const classes = useStyles();

  const handleCTAClick = useCallback(() => {
    scrollToElement(CTAId);
  }, [CTAId]);

  return (
    <>
      <Content className={classes.root}>
        <h1 className={classes.title} data-test="teaser-title">
          {intl.formatMessage(
            { id: 'CAREERS.TEASER.TITLE' },
            {
              span: (...chunks) => (
                <span className={classes.primary}>{chunks.join('')}</span>
              ),
            },
          )}
        </h1>
        <p className={classes.description} data-test="teaser-description">
          {intl.formatMessage({ id: 'CAREERS.TEASER.DESCRIPTION' })}
        </p>
        <Button
          color="primary"
          className={classes.button}
          dataTest="teaser-cta"
          onClick={handleCTAClick}
        >
          {intl.formatMessage({ id: 'CAREERS.TEASER.BUTTON' })}
        </Button>
      </Content>
      <div className={classes.image} data-test="teaser-image" />
    </>
  );
};
Teaser.propTypes = {
  CTAId: PropTypes.string.isRequired,
};

export default memo(Teaser);
