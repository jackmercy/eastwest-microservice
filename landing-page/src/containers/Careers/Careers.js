import { memo, useMemo } from 'react';
import { useIntl } from 'react-intl';
import PropTypes from 'prop-types';

import { SEO } from '../../components';
import { DEFAULT_SEO } from '../../../configs';
import { Layout, Header, Footer, Content } from '../components';

import Teaser from './Teaser';
import OurMission from './OurMission';
import CoreValues from './CoreValues';
import Benefits from './Benefits';
import Opportunities from './Opportunities';
import { MenuScrollableLayout } from './components';

import useStyles from './careers.styles';

const Careers = ({ categories, vacancies }) => {
  const intl = useIntl();
  const classes = useStyles();
  const isShownVacancySection = useMemo(
    () => categories.count > 0 && vacancies.count > 0,
    [categories, vacancies],
  );
  const ids = useMemo(
    () => ['our-mission', 'core-values', 'benefits', 'opportunities'],
    [],
  );
  const menuConfigs = useMemo(
    () => [
      {
        id: ids[0],
        label: intl.formatMessage({ id: 'CAREERS.MENU.01' }),
      },
      {
        id: ids[1],
        label: intl.formatMessage({ id: 'CAREERS.MENU.02' }),
      },
      {
        id: ids[2],
        label: intl.formatMessage({ id: 'CAREERS.MENU.03' }),
      },
      ...(isShownVacancySection
        ? [
          {
            id: ids[3],
            label: intl.formatMessage({ id: 'CAREERS.MENU.04' }),
          },
        ]
        : []),
    ],
    [intl, vacancies],
  );

  return (
    <Layout>
      <SEO {...DEFAULT_SEO.PAGES.CAREERS} />
      <Header />
      <Teaser CTAId={ids[3]} />
      <Content className={classes.menuScrollableLayout}>
        <MenuScrollableLayout menuConfigs={menuConfigs}>
          <OurMission id={ids[0]} />
          <CoreValues id={ids[1]} />
          <Benefits id={ids[2]} />
          {isShownVacancySection ? (
            <Opportunities
              id={ids[3]}
              categories={categories}
              vacancies={vacancies}
            />
          ) : null}
        </MenuScrollableLayout>
      </Content>
      <Footer />
    </Layout>
  );
};
Careers.propTypes = {
  categories: PropTypes.shape({
    count: PropTypes.number,
    data: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
      }),
    ),
  }),
  vacancies: PropTypes.shape({
    count: PropTypes.number,
    data: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        subTitle: PropTypes.string.isRequired,
        slug: PropTypes.string.isRequired,
        category: PropTypes.string.isRequired,
      }),
    ),
  }),
};
Careers.defaultProps = {
  categories: { count: 0, data: [] },
  vacancies: { count: 0, data: [] },
};

export default memo(Careers);
