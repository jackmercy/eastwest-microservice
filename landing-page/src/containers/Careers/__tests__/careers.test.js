import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import preloadAll from 'jest-next-dynamic';

import { Footer, Header } from '../../components';
import Careers from '../Careers';
import Teaser from '../Teaser';
import OurMission from '../OurMission';
import CoreValues from '../CoreValues';
import Benefits from '../Benefits';

describe('Testing Careers', () => {
  let wrapper;

  beforeAll(async () => {
    // Header and Footer included in Careers
    // And they are using dynamic import, see more in Header.js and Footer.js
    // So we need wait for it to load all before we can assert anything
    await preloadAll();
    wrapper = mount(<Careers />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<Careers />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a Header', () => {
    expect(wrapper.find(Header)).toHaveLength(1);
  });

  it('should have a Footer', () => {
    expect(wrapper.find(Footer)).toHaveLength(1);
  });

  it('should have Teaser section', () => {
    expect(wrapper.find(Teaser)).toHaveLength(1);
  });

  it('should have OurMission section', () => {
    expect(wrapper.find(OurMission)).toHaveLength(1);
  });

  it('should have CoreValues section', () => {
    expect(wrapper.find(CoreValues)).toHaveLength(1);
  });

  it('should have Benefits section', () => {
    expect(wrapper.find(Benefits)).toHaveLength(1);
  });
});
