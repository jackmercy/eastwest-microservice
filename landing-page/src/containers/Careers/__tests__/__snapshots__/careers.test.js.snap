// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Careers should render match snapshot 1`] = `
<Memo(Layout)>
  <SEO
    creator=""
    description="We are on a mission to help everyone achieve product scalability through adoption of our products and methodology. Our partners have unleashed innovation and reduced time-to-market, spending less to maintain their existing application portfolio."
    image=""
    keywords=""
    siteName=""
    title="East West - Careers"
    url="https://dev.eastwest.live/careers"
  />
  <ForwardRef(LoadableComponent) />
  <Memo(Teaser)
    CTAId="opportunities"
  />
  <Memo(Content)
    className="menuScrollableLayout-0-2-1"
  >
    <Memo(MenuScrollableLayout)
      menuConfigs={
        Array [
          Object {
            "id": "our-mission",
            "label": "Our mission",
          },
          Object {
            "id": "core-values",
            "label": "Core values",
          },
          Object {
            "id": "benefits",
            "label": "Benefits",
          },
        ]
      }
    >
      <Memo(OurMission)
        id="our-mission"
      />
      <ForwardRef(LoadableComponent)
        id="core-values"
      />
      <Memo(Benefits)
        id="benefits"
      />
    </Memo(MenuScrollableLayout)>
  </Memo(Content)>
  <ForwardRef(LoadableComponent) />
</Memo(Layout)>
`;
